vsintegration/src/FSharp.Editor/Navigation/NavigateToSearchService.fs

```diff
@@ -235,53 +118,100 @@ type internal FSharpNavigateToSearchService
         | PatternMatchKind.CamelCaseSubstring -> FSharpNavigateToMatchKind.CamelCaseSubstring
         | PatternMatchKind.CamelCaseNonContiguousSubstring -> FSharpNavigateToMatchKind.CamelCaseNonContiguousSubstring
         | PatternMatchKind.Fuzzy -> FSharpNavigateToMatchKind.Fuzzy
-        | _ -> FSharpNavigateToMatchKind.Fuzzy
+        | _ -> FSharpNavigateToMatchKind.None
+
+    let createMatcherFor searchPattern =
+        let patternMatcher =
+            lazy
+                patternMatcherFactory.CreatePatternMatcher(
+                    searchPattern,
+                    PatternMatcherCreationOptions(
+                        cultureInfo = CultureInfo.CurrentUICulture,
+                        flags =
+                            (PatternMatcherCreationFlags.AllowFuzzyMatching
+                             ||| PatternMatcherCreationFlags.AllowSimpleSubstringMatching)
```

Comment:
Is this a recommended set of flags? Are there any other interesting flags there by chance?

Reply:
fuzzy matching is a must, `AllowSimpleSubstringMatching` probably we could do without, unless it enhances fuzzy matching to substrings? This is very tersely documented unfortunately. 
`IncludeMatchedSpans` would give us that nice highlighting of matching substrings, but we would have to expose this functionality in `IFSharpNavigateToSearchService` on Roslyn side first.

Reply:
Thanks for looking into this. 

> IncludeMatchedSpans would give us that nice highlighting of matching substrings, but we would have to expose this functionality in IFSharpNavigateToSearchService on Roslyn side first.

Sounds helpful. I think we can do this as a followup and, if so, even bring more attention to this - you are turning this semi-broken functionality into something we can be proud of.

Reply:
Actually there is one low hanging fruit. C# lang service has a feature to search by namespace/container like this:
![image](https://user-images.githubusercontent.com/1760221/222370668-640d6410-f2aa-42b9-882a-2338b9906e26.png)

I think we could unlock it quite easily for F#.

Reply:
Yep, sounds good.